### PR TITLE
[Sources] Snowflake - Part 3

### PR DESCRIPTION
**Pull request by vellames**
Part 3 of #5226 

- Deleted User
- Deleted Role

### <samp>🤖 Generated by Copilot at c60f6ed</samp>

This pull request adds two new sources to the snowflake component that emit events when a role or a user is deleted from snowflake. It also updates the snowflake package version and the `pnpm-lock.yaml` file. The new sources use a common base component from `common-delete.mjs` that handles the common logic for delete sources.

### <samp>🤖 Generated by Copilot at c60f6ed</samp>

> _The snowflake package got a bump_
> _To add some features and fix some slump_
> _With common-delete they made a source_
> _For role and user, of course_
> _And pnpm-lock kept things in a clump_

### <samp>🤖 Generated by Copilot at c60f6ed</samp>

*  Update snowflake package version to 0.7.0 ([link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-9f1a179ed7be32b74e2e4b5d5b1158fc619865e5629751bc262062e95ca7760bL3-R3))
*  Add common delete source component for snowflake objects ([link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-22f2059ea1a6c0e051d68e9afdd2242ac1eab7b55389316e430d8a61aaad5cc0R1-R83))
*  Extend common delete source component for deleted role source ([link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-910501cfc3cf214f00dbefc38dda8f41eb9851a2dfdc011c73bcc0bd980f2458R1-R17))
*  Extend common delete source component for deleted user source ([link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-af028735241aa73474fff8bc62bcd5e49f8cd2301fbe982ed954f893084380fcR1-R17))
*  Add, remove, or update entries for various components in `pnpm-lock.yaml` file ([link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR401-R403), [link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1320-R1322), [link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1860-R1862), [link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1906-R1908), [link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL2836-R2849), [link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2966-R2968), [link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3514-R3516), [link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3695-R3697), [link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3755-R3757), [link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3952-R3954), [link](https://github.com/PipedreamHQ/pipedream/pull/6486/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4470-R4472))

### CODE REVIEW

File: components/snowflake/sources/common-delete.mjs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+import { v4 as uuid } from "uuid";
+
+export default {
+  type: "source",
+  props: {
+    snowflake,
+    db: "$.service.db",
+    timer: {
+      description: "Watch for changes on this schedule",
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/snowflake/sources/common-delete.mjs:11]: Changed 'Watch' to 'Monitor' to follow Google's style guide (use active voice)

File: components/snowflake/sources/deleted-role/deleted-role.mjs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import common from "../common-delete.mjs";
+
+export default {
+  ...common,
+  type: "source",
+  key: "snowflake-deleted-role",
+  name: "New Deleted Role",
+  description: "Emit new event when a role is deleted",
```
Comment (dylburger) [inline components/snowflake/sources/deleted-role/deleted-role.mjs:8]:
```suggestion
  description: "Emit a new event when a role is deleted",
```

File: components/snowflake/sources/deleted-user/deleted-user.mjs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import common from "../common-delete.mjs";
+
+export default {
+  ...common,
+  type: "source",
+  key: "snowflake-deleted-user",
+  name: "New Deleted User",
+  description: "Emit new event when a user is deleted",
```
Comment (dylburger) [inline components/snowflake/sources/deleted-user/deleted-user.mjs:8]:
```suggestion
  description: "Emit a new event when a user is deleted",
```

File: components/snowflake/sources/common-delete.mjs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+import { v4 as uuid } from "uuid";
+
+export default {
+  type: "source",
+  props: {
+    snowflake,
+    db: "$.service.db",
+    timer: {
+      description: "Monitor for changes on this schedule",
```
Comment (dylburger) [inline components/snowflake/sources/common-delete.mjs:11]:
```suggestion
      description: "Monitor for changes on this schedule.",
```

Comment (dylburger) [reply components/snowflake/sources/common-delete.mjs:11]: Added a period at the end of the sentence for consistency with Google's Developer docs style guide.

File: components/snowflake/sources/common-delete.mjs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+import { v4 as uuid } from "uuid";
+
+export default {
+  type: "source",
+  props: {
+    snowflake,
+    db: "$.service.db",
+    timer: {
+      description: "Monitor for changes on this schedule.",
```
Comment (dylburger) [inline components/snowflake/sources/common-delete.mjs:11]:
```suggestion
      description: "Monitor for changes on this schedule.",
```

Comment (dylburger) [reply components/snowflake/sources/common-delete.mjs:11]: No changes needed

**Comment by vellames** [comment]
Visual Ping was updated by linter. No real code was made there

**Review by luancazarine — APPROVED** [review]
Hi @vellames, thanks for your contribution! LGTM!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Snowflake_Part_3_6486_1254.pdf](https://github.com/PipedreamHQ/pipedream/files/11504801/Snowflake_Part_3_6486_1254.pdf)

**Pull request merged**